src/pages/dev/general-message-passing/overview.mdx

```diff
@@ -21,14 +21,52 @@ With GMP you can:
+
+
+### Real-World Breakdown of Cross-Chain [Contract Calls](https://axelarscan.io/gmp/0x93cb0b614b07d6050b164cc3e35da617a2fbefc13069a35369894cac74b861a2:54) 
+
+![simple-gmp-diagram.png](/images/simple-gmp-diagram.png)
+
+#### Setup
+
+a. The destination application contract implements the `IAxelarExecutable.sol` interface to receive the cross-chain message.
+
+b. The destination application contract stores the address of the destination chain's Axelar gateway contract.
+
+#### On Source Chain
+1. A smart contract on the source chain [calls](https://moonscan.io/tx/0x93cb0b614b07d6050b164cc3e35da617a2fbefc13069a35369894cac74b861a2) `AxelarGateway.callContract()` with the destination chain/address and `payload`, which emits the `ContractCall` event.
+
+2. The smart contract can deposit tokens to the [`AxelarGasService`](https://github.com/axelarnetwork/axelar-cgp-solidity/blob/main/contracts/gas-service/AxelarGasService.sol#L122) contract in the same transaction to pay the Axelar relayers for submitting the intermediate transactions required for cross-chain execution.
+
+#### On Axelar Network
+
+3. A [relayer](https://docs.axelar.dev/learn/network/flow#message-processing-and-relayers) monitors the `ContractCall` event and submits a transaction to the Axelar network to request validation. The relayer also stores the `payload` in a database, keyed by `hash(payload)` for later retrieval.
+
+4. Axelar validators then [vote](https://axelarscan.io/evm-poll/434420) on-chain to validate the `ContractCall` event content.
+
+5. A relayer requests the Axelar network to prepare a command batch, including the pending payload approval (potentially batched with other messages), and requests validator signatures.
+
+6. A signed [batch](https://axelarscan.io/evm-batches?commandId=0x47d0de91330856d70caecf442341be3faf6e644b83892b214c5a2bcc673ba8ca) of approved payloads is prepared on Axelar that anyone can view.
+
+#### On Destination Chain
+
+7. A relayer [submits](https://bscscan.com/tx/0x72e6c040bfbf26073cdcf55cc4db571badcadd3b9316cf0f53b72f980d3e5100) the signed batch to the destination gateway contract, which records the [approval](https://github.com/axelarnetwork/cgp-spec/blob/main/solidity/contracts/AxelarGateway.sol#L109) of the payload hash and emits a `ContractCallApproved` [event](https://github.com/axelarnetwork/cgp-spec/blob/main/solidity/contracts/AxelarGateway.sol#L144).
+
+8. A trustless relayer service (can be anyone) listens for this event and [calls](https://bscscan.com/tx/0x24886831c6348f036be26193d3fd74f2a08b9b9c10cae4e4bb99677687d8b71f) `IAxelarExecutable.execute()` on the destination contract with the `payload` and other data as params.
+
+9. The `execute` method on the destination [contract](https://github.com/axelarnetwork/cgp-spec/blob/main/solidity/contracts/executable/AxelarExecutable.sol#L18) verifies the call was indeed approved by Axelar validators by calling `validateContractCall()` on its Axelar gateway contract.
```

Comment:
maybe mention our contract inherits from the AxelarExecutable which defines execute/executeWithToken contract and in that contract which we inherit from is where `validateContractCall()` is defined

Reply:
Updated. 

FYI: In that step, the Link referenced as the destination contract points to the AxelarExecutable file.